core/tx_pool.go

```diff
@@ -496,6 +496,40 @@ func (pool *TxPool) stats() (int, int) {
 	return pending, queued
 }
 
+// StatsWithMinBaseFee retrieves the current pool stats, namely the number of pending and the
+// number of queued (non-executable) transactions greater equal minBaseFee.
+func (pool *TxPool) StatsWithMinBaseFee(minBaseFee *big.Int) (int, int) {
+	pool.mu.RLock()
```

Comment:
Any concern that this might delay the `worker` or delay tx insertion?

Reply:
I think it should be ok as we do this only once per block on RPC nodes. The result is cached in the GPO. 
Also some of the work we're doing is useful, as we use `flatten` which populates the cache of the `txlist`. This is also used when calling `PendingWithMax`.